src/psyclone/core/symbolic_maths.py

```diff
@@ -82,13 +82,18 @@ def get():
 
     # -------------------------------------------------------------------------
     @staticmethod
-    def equal(exp1, exp2):
+    def equal(exp1, exp2, assume=None):
         '''Test if the two PSyIR expressions are symbolically equivalent.
+        The optional assume dictionary can contain information which variables
+        are known to be the same. For example, if `assume={'i': 'j'}`, then
```

Comment:
I don't think 'assume' is quite the right name here. Perhaps "identical_variables" or something?